crates/deno-subsystem/deno-model-builder/src/module_skeleton_generator.rs

```diff
@@ -436,3 +494,203 @@ fn typescript_base_type(exo_type_name: &str) -> String {
+
+        assert_eq!(generated_code, expected_code);
+    }
+
+    #[test]
+    fn test_generates_module_definitions_correctly() {
+        use std::fs;
+        use core_plugin_interface::core_model_builder::ast::ast_types::{AstModel, AstField, AstFieldType, AstModule, AstModelKind};
+        use codemap::CodeMap;
+        use std::path::PathBuf;
+
+        let span = CodeMap::new().add_file("".to_string(), "".to_string()).span.subspan(0, 0);
+        
+        let module = AstModule {
+            name: "TestModule".to_string(),
+            types: vec![
+                AstModel {
+                    name: "TestType1".to_string(),
+                    kind: AstModelKind::Type,
+                    fields: vec![
+                        AstField {
+                            name: "field1".to_string(),
+                            typ: AstFieldType::Plain("String".to_string(), vec![], true, span),
+                            annotations: Default::default(),
+                            default_value: None,
+                            span,
+                        }
+                    ],
+                    annotations: Default::default(),
+                    span,
+                },
+                AstModel {
+                    name: "TestType2".to_string(),
+                    kind: AstModelKind::Type,
+                    fields: vec![
+                        AstField {
+                            name: "field2".to_string(),
+                            typ: AstFieldType::Plain("Number".to_string(), vec![], true, span),
+                            annotations: Default::default(),
+                            default_value: None,
+                            span,
+                        }
+                    ],
+                    annotations: Default::default(),
+                    span,
+                }
+            ],
+            annotations: Default::default(),
+            base_exofile: PathBuf::new(),
+            interceptors: vec![],
+            methods: vec![],
+            span,
+        };
+
+        let generated_dir = Path::new("generated");
+
+        // Ensure the directory does not exist before the test
+        if generated_dir.exists() {
+            fs::remove_dir_all(generated_dir).unwrap();
+        }
+
+        generate_module_definitions(&module).unwrap();
+
+        let module_file = generated_dir.join("TestModule.d.ts");
+        assert!(module_file.exists(), "Module {} doesn't exist", module_file.display());
+
+        let content = fs::read_to_string(module_file).unwrap();
+
+        let expected_type1 = "export interface TestType1 {\n\tfield1: string\n}";
+        let expected_type2 = "export interface TestType2 {\n\tfield2: Number\n}";
+        assert!(content.contains(expected_type1), "TestType1 não encontrado");
+        assert!(content.contains(expected_type2), "TestType2 não encontrado");
```

Comment:
May be just add `content = expected_type1 + "\n" + expected_type2` and assert file context against it. That will ensure, for example, that the type definitions aren't repeated in the same file.